Realm/RLMObject.mm

```diff
@@ -40,7 +40,11 @@ - (instancetype)init {
 }
 
 - (instancetype)initWithValue:(id)value {
-    return [super initWithValue:value schema:RLMSchema.sharedSchema];
+    RLMSchema *schema = RLMSchema.partialSharedSchema;
+    if (![schema schemaForClassName:self.class.className]) {
```

Comment:
@jpsim not entirely possible, because we need to ensure that classes referenced in link properties are loaded at the same time as the classes that reference them -- this is why classes are batch registered, rather than doing it one-by-one.